### PR TITLE
Spirv opt uses optimizer class

### PR DESCRIPTION
**Pull request by dneto0**
This builds on https://github.com/KhronosGroup/SPIRV-Tools/pull/612

### CODE REVIEW

File: tools/opt/opt.cpp [hunk 1]
```diff
@@ -13,16 +13,16 @@
 // limitations under the License.
 
 #include <cstring>
+#include <algorithm>
```
Comment (antiagainst) [inline tools/opt/opt.cpp]: Order

File: test/opt/optimizer_test.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#include <gmock/gmock.h>
+
+#include "spirv-tools/libspirv.hpp"
+#include "spirv-tools/optimizer.hpp"
+
+#include "pass_fixture.h"
+
+namespace {
+
+using spvtools::CreateStripDebugInfoPass;
```
Comment (antiagainst) [inline test/opt/optimizer_test.cpp]: Order

File: test/opt/optimizer_test.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+  Optimizer opt(SPV_ENV_UNIVERSAL_1_0);
+  opt.RegisterPass(CreateNullPass());
+  auto orig_size = binary.size();
+  // Now change the size.  Add a word that will be ignored
+  // by the optimizer.
+  binary.push_back(42);
+  EXPECT_THAT(orig_size + 1, Eq(binary.size()));
+  opt.Run(binary.data(), orig_size, &binary);  // This is the key.
+  // The binary vector should have been rewritten.
```
Comment (antiagainst) [inline test/opt/optimizer_test.cpp]: Nice!

File: tools/opt/opt.cpp (hunk 1 above)
Comment (dneto0) [inline tools/opt/opt.cpp]: C system files go before C++ files

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

:-)

**Comment by dneto0** [comment]
Thanks for approval.  I'll fix the order of the using decl, keep approval. Then I'll submit after the bots come back green.

**Comment by dneto0** [comment]
Rebased. Keeping approval. No need for review.  :-)

File: tools/opt/opt.cpp (hunk 1 above)
Comment (antiagainst) [inline tools/opt/opt.cpp]: TIrL! (Today I reLearned!)

**Pull request merged**